codesandbox/codesandbox-client
# Convert all Medium blogposts to Markdown

## What kind of change does this PR introduce?
I extracted all of @CompuIves' [Medium posts](https://medium.com/@CompuIves/latest) into Markdown files (just like @kentcdodds did in his [three](https://www.youtube.com/watch?v=9Uy1db7qsT0&list=PLV5CVI1eNcJgJCEkMlsqXea6OIF_uV_ub) [part](https://www.youtube.com/watch?v=T70G-kJonXI&list=PLV5CVI1eNcJgJCEkMlsqXea6OIF_uV_ub) [tutorial](https://www.youtube.com/watch?v=eq7VuQnAzAE&list=PLV5CVI1eNcJgJCEkMlsqXea6OIF_uV_ub)).

The main reason for doing this, was the fact that right now we only see 8 Medium posts + the newly created investor seed post.  
This comes from the fact that we fetch all Medium posts from https://medium.com/feed/@CompuIves, which only return the last 10 created items of @CompuIves (including comments, which he has 2 of).

I've looked into alternatives to fetch all Medium posts and found that you can get them all sorted from https://medium.com/@CompuIves/latest?format=json in a (broken) JSON format.  
The only problem here is that you don't get the content of the post itself.  
These posts were also limited to the latest 10 articles btw.

### The only things that I still need to look at are
- [ ] Delete our own copy of @kentcdodds' `remark-embedder` plugin when that's extracted from [kentcdodds/kentcdodds.com](https://github.com/kentcdodds/kentcdodds.com) (see kentcdodds/kentcdodds.com#172)
- [x] Decide what we want to do with all GitHub gists
  - ~~Embed them? (This will need a change in the `remark-embedder` plugin)~~
  - Convert them to code blocks inside the Markdown file? ✅
- [x] Decide what to do with the movie from [`announcing-codesandbox-2-0`](https://hackernoon.com/announcing-codesandbox-2-0-938cff3a0fcb)
  - Convert it to GIF ✅
  - ~~Embed it~~
  - ~~Leave it out~~
- [x] Checkout how we can fix inline links that show the link themself as text (this needs to be solved by the `remark-embedder` plugin), since that now shows the embed instead of the text or it's a broken link (see kentcdodds/kentcdodds.com#194)
  ```md
  [codesandbox.io/s/vue](https://codesandbox.io/s/vue)
  [https://codesandbox.io/s/vue](https://codesandbox.io/s/vue)
  ```
- [x] Revert 40a986f1, since the Medium posts are still used on the homepage.
  Will create a separate PR to delete them there and cleanup the config
- [x] Optimize all images using [tinypng.com](https://tinypng.com) & [gifcompressor.com](https://gifcompressor.com)
- [ ] Decide what to do with all the image (and sandbox) descriptions
  - Leave them out
  - Show them too
- [x] Fix `remark-embedder` plugin for Twitter accounts with 2 underscores (`_`) (see kentcdodds/kentcdodds.com#191)
  ```md
  https://twitter.com/brian_d_vaughn/status/901488766581227520
  ```
  **Edit**: needs further investigation from ourselves, since [it's not the `remark-embedder` plugin](https://github.com/kentcdodds/kentcdodds.com/issues/191#issuecomment-499179350).
  **Edit 2**: It seems to be a `gatsby-transformer-remark` problem (see gatsbyjs/gatsby#14715).
  **Edit 3**: `gatsby-transformer-remark`'s `pedantic` option should be set to `false` (see https://github.com/gatsbyjs/gatsby/issues/14715#issuecomment-502871943)

### Follow-ups
- [ ] Change `/post/blog-slug` to `/blog/blog-slug` an redirect all old links to the new ones
- [ ] Make the blog-section on the homepage work with the new Markdown files instead of only the Medium posts
- [ ] Use the `remark-embedder` plugin in other parts (docs, ...) too (delete the `https://codesandbox.io/s/embed/...` links)

## What is the current behavior?
Having both articles from self-written Markdown files and fetched the last 8 articles from Medium

## What is the new behavior?
All articles are "self-written" Markdown files and include all of @CompuIves' articles that are present on his Medium profile.

## What steps did you take to test this?
- Run `yarn start:home`
- Go to `localhost:8000/blog` and see all articles are present
- Click on each article and see it's present

## Checklist
- [x] Added myself to contributors table
